web/src/engine/websites/ComicWalker.ts

```diff
@@ -0,0 +1,113 @@
+import { Tags } from '../Tags';
+import icon from './ComicWalker.webp';
+import { type Chapter, DecoratableMangaScraper, type Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { Fetch, FetchCSS, FetchJSON, FetchRequest } from '../FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+
+type TEndpoints = {
+    nc: string,
+    cw: string
+}
+
+type APIPages = {
+    data: {
+        result: {
+            meta: {
+                source_url: string,
+                drm_hash: string
+            }
+        }[]
+    }
+}
+
+function ChapterExtractor(anchor: HTMLAnchorElement) {
+    return {
+        id: anchor.pathname + anchor.search,
+        title: anchor.title.trim()
+    };
+}
+
+@Common.MangaCSS(/^https?:\/\/comic-walker\.com\/contents\/detail\/[^/]+\/$/, 'div#mainContent div#detailIndex div.comicIndex-box h1')
+@Common.ChaptersSinglePageCSS('div#ulreversible ul#reversible li a', ChapterExtractor)
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('comicwalker', `コミックウォーカー (ComicWalker)`, 'https://comic-walker.com', Tags.Language.Multilingual, Tags.Source.Official, Tags.Media.Manga);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangasList : Manga[] = [];
+        for (const language of ['en', 'tw', 'jp']) {
+            await this.setLanguage(language);
+            const mangas = await Common.FetchMangasMultiPageCSS.call(this, provider, '/contents/list/p?={page}', 'div.comicPage ul.tileList li a', 1, 1, 0, Common.AnchorInfoExtractor(false, '.contents_tile_epi'));
+            mangasList.push(...mangas);
+        }
+        return mangasList;
+    }
+
+    private async setLanguage(language: string): Promise<void> {
+        const request = new FetchRequest(`${this.URI.origin}/set_lang/${ language }/`);
+        await Fetch(request);
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        //get endpoints data
+        const request = new FetchRequest(new URL(chapter.Identifier, this.URI).href);
+        const mainApp = await FetchCSS(request, 'main#app');
+        const endpoints: string | TEndpoints = mainApp[0].dataset.apiEndpointUrl ? mainApp[0].dataset.apiEndpointUrl : JSON.parse(mainApp[0].dataset.apiEndpointUrls);
+        const uri = `${(endpoints as TEndpoints).nc || (endpoints as TEndpoints).cw || endpoints}/api/v1/comicwalker/episodes/${mainApp[0].dataset.episodeId}/frames`;
+
+        const { data } = await FetchJSON<APIPages>(new FetchRequest(uri));
```

Comment:
Will this work if a language is set which is incompatible with the chapter?
e.g., [KDCW_MF09203324020032_68](https://comic-walker.com/viewer/?tw=2&dlcl=en&cid=KDCW_MF09203324020032_68) is only available in english, what happens when last language was set to `jp` (after manga list update)?

Reply:
Right, got that fixed properly